### ISSUE TITLE
[BUG] cudf.read_csv failures with managed_memory=True

### ISSUE THREAD

**Issue by afender** [rapidsai/cudf#5190]
**Describe the bug**
I got `cudf.read_csv`  failures when trying to use UVM as shown below. Some inputs do not trigger this issue.

**Steps/Code to reproduce bug**
CSV file : https://drive.google.com/open?id=1w3L4IUeofv5_iNFaT6qie8nZQReVnNm-
This dataset is less than 1% of the device memory.

```python
import rmm
import cudf
rmm.reinitialize(managed_memory=True)
gdf = cudf.read_csv('large.csv', delimiter='\t', dtype=['int32', 'int32'])
print(gdf)
```

returns

```
terminate called after throwing an instance of 'thrust::system::system_error'
  what():  parallel_for failed: cudaErrorIllegalAddress: an illegal memory access was encountered
Aborted (core dumped)
```
Everything is fine when `managed_memory=False`

I also got very strange behaviors while experimenting with other configs. For instance `pool_allocator=True, managed_memory=True` would pass only if `initial_pool_size` is large.

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda (nightly)

**Environment details**
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 451edb18e1838c5c3045e71f7ca0d1470fb16f70 (HEAD -> branch-0.14, origin/branch-0.14, origin/HEAD)
     Merge: eeff08a59 390a8e723
     Author: Conor Hoekstra <36027403+codereport@users.noreply.github.com>
     Date:   Wed May 13 16:02:52 2020 -0400
     
     Merge pull request #4822 from codereport/fea-raii-pq-chunked
     
     [REVIEW] RAII-ify `pq_chunked_state` [skip ci]
     **git submodules***
     -b165e1fb11eeea64ccf95053e40f2424312599cc thirdparty/cub
     -50680b29d1f9bc41905acf006a1a94a40e0728e1 thirdparty/jitify
     -cdcda484d0c7db114ea29c3b33429de5756ecfd8 thirdparty/libcudacxx
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=18.04
     DISTRIB_CODENAME=bionic
     DISTRIB_DESCRIPTION="Ubuntu 18.04.4 LTS"
     NAME="Ubuntu"
     VERSION="18.04.4 LTS (Bionic Beaver)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 18.04.4 LTS"
     VERSION_ID="18.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=bionic
     UBUNTU_CODENAME=bionic
     Linux afhpz8 5.3.0-51-generic #44~18.04.2-Ubuntu SMP Thu Apr 23 14:27:18 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Wed May 13 19:37:44 2020
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 440.64.00    Driver Version: 440.64.00    CUDA Version: 10.2     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |===============================+======================+======================|
     |   0  Quadro RTX 8000     On   | 00000000:15:00.0 Off |                  Off |
     | 33%   42C    P8    42W / 260W |     56MiB / 48601MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     |   1  Quadro RTX 8000     On   | 00000000:2D:00.0  On |                  Off |
     | 34%   49C    P8    27W / 260W |    952MiB / 48592MiB |      0%      Default |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                       GPU Memory |
     |  GPU       PID   Type   Process name                             Usage      |
     |=============================================================================|
     |    1      1081      G   /usr/lib/xorg/Xorg                            71MiB |
     |    1      1204      G   /usr/bin/gnome-shell                          17MiB |
     |    1      1517      G   /usr/lib/xorg/Xorg                           321MiB |
     |    1      1649      G   /usr/bin/gnome-shell                         286MiB |
     |    1      3164      G   ...quest-channel-token=6818357383536296573    84MiB |
     |    1     11925      G   ...quest-channel-token=7217908864477463812   154MiB |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:        x86_64
     CPU op-mode(s):      32-bit, 64-bit
     Byte Order:          Little Endian
     CPU(s):              12
     On-line CPU(s) list: 0-11
     Thread(s) per core:  2
     Core(s) per socket:  6
     Socket(s):           1
     NUMA node(s):        1
     Vendor ID:           GenuineIntel
     CPU family:          6
     Model:               85
     Model name:          Intel(R) Xeon(R) Gold 6128 CPU @ 3.40GHz
     Stepping:            4
     CPU MHz:             1200.190
     CPU max MHz:         3700.0000
     CPU min MHz:         1200.0000
     BogoMIPS:            6800.00
     Virtualization:      VT-x
     L1d cache:           32K
     L1i cache:           32K
     L2 cache:            1024K
     L3 cache:            19712K
     NUMA node0 CPU(s):   0-11
     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke md_clear flush_l1d
     
     ***CMake***
     /home/afender/miniconda3/envs/cugraph_dev/bin/cmake
     cmake version 3.17.0
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
     Copyright (C) 2017 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2019 NVIDIA Corporation
     Built on Wed_Oct_23_19:24:38_PDT_2019
     Cuda compilation tools, release 10.2, V10.2.89
     
     ***Python***
     /home/afender/miniconda3/envs/cugraph_dev/bin/python
     Python 3.7.6
     
     ***Environment Variables***
     PATH                            : /home/afender/miniconda3/envs/cugraph_dev/bin:/home/afender/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/usr/local/cuda/bin:/home/afender/ompi/build/bin:/home/afender/ucx/build/bin
     LD_LIBRARY_PATH                 : :/usr/local/cuda/lib64:/home/afender/nccl/build/build/lib:/home/afender/ompi/build/lib:/home/afender/ucx/build/bin
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/afender/miniconda3/envs/cugraph_dev
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/afender/miniconda3/condabin/conda
     # packages in environment at /home/afender/miniconda3/envs/cugraph_dev:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     _openmp_mutex             4.5                      1_llvm    conda-forge
     alabaster                 0.7.12                     py_0    conda-forge
     arrow-cpp                 0.15.0           py37h090bef1_2    conda-forge
     astroid                   2.3.3                    py37_1    conda-forge
     attrs                     19.3.0                     py_0    conda-forge
     babel                     2.8.0                      py_0    conda-forge
     backcall                  0.1.0                      py_0    conda-forge
     bleach                    3.1.5              pyh9f0ad1d_0    conda-forge
     bokeh                     2.0.1            py37hc8dfbb8_0    conda-forge
     boost                     1.70.0           py37h9de70de_1    conda-forge
     boost-cpp                 1.70.0               h8e57a91_2    conda-forge
     brotli                    1.0.7             he1b5a44_1001    conda-forge
     brotlipy                  0.7.0           py37h8f50634_1000    conda-forge
     bzip2                     1.0.8                h516909a_2    conda-forge
     c-ares                    1.15.0            h516909a_1001    conda-forge
     ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
     certifi                   2020.4.5.1       py37hc8dfbb8_0    conda-forge
     cffi                      1.14.0           py37hd463f26_0    conda-forge
     chardet                   3.0.4           py37hc8dfbb8_1006    conda-forge
     click                     7.1.2              pyh9f0ad1d_0    conda-forge
     cloudpickle               1.4.1                      py_0    conda-forge
     cmake                     3.17.0               h28c56e5_0    conda-forge
     commonmark                0.9.1                      py_0    conda-forge
     cryptography              2.9.2            py37hb09aad4_0    conda-forge
     cudatoolkit               10.2.89              h6bb024c_0    nvidia
     cudf                      0.14.0a200513         py37_4605    rapidsai-nightly
     cudnn                     7.6.5                cuda10.2_0
     cugraph                   0.13.0a0+145.g5e9cb431          pypi_0    pypi
     cupy                      7.4.0            py37h940342b_2    conda-forge
     cython                    0.29.17          py37h3340039_0    conda-forge
     cytoolz                   0.10.1           py37h516909a_0    conda-forge
     dask                      2.16.0                     py_0    conda-forge
     dask-core                 2.16.0                     py_0    conda-forge
     dask-cuda                 0.14.0a200513          py37_102    rapidsai-nightly
     dask-cudf                 0.14.0a200513         py37_4605    rapidsai-nightly
     decorator                 4.4.2                      py_0    conda-forge
     defusedxml                0.6.0                      py_0    conda-forge
     distributed               2.16.0           py37hc8dfbb8_0    conda-forge
     dlpack                    0.2                  he1b5a44_1    conda-forge
     docutils                  0.16             py37hc8dfbb8_1    conda-forge
     double-conversion         3.1.5                he1b5a44_2    conda-forge
     entrypoints               0.3             py37hc8dfbb8_1001    conda-forge
     expat                     2.2.9                he1b5a44_2    conda-forge
     fastavro                  0.23.3           py37h8f50634_0    conda-forge
     fastrlock                 0.4             py37h3340039_1001    conda-forge
     freetype                  2.10.2               he06d7ca_0    conda-forge
     fsspec                    0.7.3                      py_0    conda-forge
     future                    0.18.2           py37hc8dfbb8_1    conda-forge
     gflags                    2.2.2             he1b5a44_1002    conda-forge
     glog                      0.4.0                h49b9bf7_3    conda-forge
     grpc-cpp                  1.23.0               h18db393_0    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     icu                       64.2                 he1b5a44_1    conda-forge
     idna                      2.9                        py_1    conda-forge
     imagesize                 1.2.0                      py_0    conda-forge
     importlib-metadata        1.6.0            py37hc8dfbb8_0    conda-forge
     importlib_metadata        1.6.0                         0    conda-forge
     ipykernel                 5.2.1            py37h43977f1_0    conda-forge
     ipython                   7.14.0           py37hc8dfbb8_0    conda-forge
     ipython_genutils          0.2.0                      py_1    conda-forge
     isort                     4.3.21           py37hc8dfbb8_1    conda-forge
     jedi                      0.17.0           py37hc8dfbb8_0    conda-forge
     jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
     jpeg                      9c                h14c3975_1001    conda-forge
     jsonschema                3.2.0            py37hc8dfbb8_1    conda-forge
     jupyter_client            6.1.3                      py_0    conda-forge
     jupyter_core              4.6.3            py37hc8dfbb8_1    conda-forge
     krb5                      1.17.1               h2fd8d38_0    conda-forge
     lazy-object-proxy         1.4.3            py37h8f50634_2    conda-forge
     ld_impl_linux-64          2.34                 h53a641e_0    conda-forge
     libblas                   3.8.0               16_openblas    conda-forge
     libcblas                  3.8.0               16_openblas    conda-forge
     libcudf                   0.14.0a200513     cuda10.2_4605    rapidsai-nightly
     libcugraph                0.14.0a200511      cuda10.2_517    rapidsai-nightly
     libcurl                   7.69.1               hf7181ac_0    conda-forge
     libcypher-parser          0.6.2                         1    rapidsai
     libedit                   3.1.20170329      hf8c457e_1001    conda-forge
     libevent                  2.1.10               h72c5cf5_0    conda-forge
     libffi                    3.2.1             he1b5a44_1007    conda-forge
     libgcc-ng                 9.2.0                h24d8f2e_2    conda-forge
     libgfortran-ng            7.3.0                hdf63c60_5    conda-forge
     liblapack                 3.8.0               16_openblas    conda-forge
     libllvm8                  8.0.1                hc9558a2_0    conda-forge
     libnvstrings              0.14.0a200513     cuda10.2_4605    rapidsai-nightly
     libopenblas               0.3.9                h5ec1e0e_0    conda-forge
     libpng                    1.6.37               hed695b0_1    conda-forge
     libprotobuf               3.8.0                h8b12597_0    conda-forge
     librmm                    0.14.0a200513      cuda10.2_340    rapidsai-nightly
     libsodium                 1.0.17               h516909a_0    conda-forge
     libssh2                   1.8.2                h22169c7_2    conda-forge
     libstdcxx-ng              9.2.0                hdf63c60_2    conda-forge
     libtiff                   4.1.0                hfc65ed5_0    conda-forge
     libuv                     1.34.0               h516909a_0    conda-forge
     llvm-openmp               10.0.0               hc9558a2_0    conda-forge
     llvmlite                  0.32.0           py37h5202443_0    conda-forge
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.8.3             he1b5a44_1001    conda-forge
     markdown                  3.0.1                    pypi_0    pypi
     markupsafe                1.1.1            py37h8f50634_1    conda-forge
     mccabe                    0.6.1                      py_1    conda-forge
     mistune                   0.8.4           py37h8f50634_1001    conda-forge
     more-itertools            8.2.0                      py_0    conda-forge
     msgpack-python            1.0.0            py37h99015e2_1    conda-forge
     nbconvert                 5.6.1            py37hc8dfbb8_1    conda-forge
     nbformat                  5.0.6                      py_0    conda-forge
     nbsphinx                  0.7.0              pyh9f0ad1d_0    conda-forge
     nccl                      2.6.4.1              hc6a2c23_0    conda-forge
     ncurses                   6.1               hf484d3e_1002    conda-forge
     networkx                  2.4                        py_1    conda-forge
     notebook                  6.0.3            py37hc8dfbb8_0    conda-forge
     numba                     0.49.0           py37h0573a6f_0
     numpy                     1.18.4           py37h8960a57_0    conda-forge
     numpydoc                  0.9.2                      py_0    conda-forge
     nvstrings                 0.14.0a200513         py37_4605    rapidsai-nightly
     olefile                   0.46                       py_0    conda-forge
     openssl                   1.1.1g               h516909a_0    conda-forge
     packaging                 20.1                       py_0    conda-forge
     pandas                    0.25.3           py37hb3f55d8_0    conda-forge
     pandoc                    2.9.2.1                       0    conda-forge
     pandocfilters             1.4.2                      py_1    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     parso                     0.7.0              pyh9f0ad1d_0    conda-forge
     partd                     1.1.0                      py_0    conda-forge
     pexpect                   4.8.0            py37hc8dfbb8_1    conda-forge
     pickleshare               0.7.5           py37hc8dfbb8_1001    conda-forge
     pillow                    7.1.2            py37h718be6c_0    conda-forge
     pip                       20.1               pyh9f0ad1d_0    conda-forge
     pluggy                    0.13.1           py37hc8dfbb8_1    conda-forge
     prometheus_client         0.7.1                      py_0    conda-forge
     prompt-toolkit            3.0.5                      py_0    conda-forge
     psutil                    5.7.0            py37h8f50634_1    conda-forge
     ptyprocess                0.6.0                   py_1001    conda-forge
     py                        1.8.1                      py_0    conda-forge
     pyarrow                   0.15.0           py37h8b68381_1    conda-forge
     pycparser                 2.20                       py_0    conda-forge
     pygments                  2.6.1                      py_0    conda-forge
     pylint                    2.4.4                    py37_0
     pynvml                    8.0.4                      py_0    conda-forge
     pyopenssl                 19.1.0                     py_1    conda-forge
     pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
     pyrsistent                0.16.0           py37h8f50634_0    conda-forge
     pysocks                   1.7.1            py37hc8dfbb8_1    conda-forge
     pytest                    5.4.2            py37hc8dfbb8_0    conda-forge
     python                    3.7.6           h8356626_5_cpython    conda-forge
     python-dateutil           2.8.1                      py_0    conda-forge
     python-louvain            0.13                       py_0    conda-forge
     python_abi                3.7                     1_cp37m    conda-forge
     pytz                      2020.1             pyh9f0ad1d_0    conda-forge
     pyyaml                    5.3.1            py37h8f50634_0    conda-forge
     pyzmq                     19.0.1           py37hac76be4_0    conda-forge
     re2                       2020.05.01           he1b5a44_0    conda-forge
     readline                  8.0                  hf8c457e_0    conda-forge
     recommonmark              0.6.0                      py_0    conda-forge
     requests                  2.23.0             pyh8c360ce_2    conda-forge
     rhash                     1.3.6             h14c3975_1001    conda-forge
     rmm                       0.14.0a200513          py37_340    rapidsai-nightly
     scipy                     1.4.1            py37ha3d9a3c_3    conda-forge
     send2trash                1.5.0                      py_0    conda-forge
     setuptools                46.3.0           py37hc8dfbb8_0    conda-forge
     six                       1.14.0                     py_1    conda-forge
     snappy                    1.1.8                he1b5a44_1    conda-forge
     snowballstemmer           2.0.0                      py_0    conda-forge
     sortedcontainers          2.1.0                      py_0    conda-forge
     spdlog                    1.5.0                hc9558a2_0    conda-forge
     sphinx                    3.0.3                      py_0    conda-forge
     sphinx-markdown-tables    0.0.14                   pypi_0    pypi
     sphinx_rtd_theme          0.4.3                      py_0    conda-forge
     sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
     sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
     sphinxcontrib-htmlhelp    1.0.3                      py_0    conda-forge
     sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
     sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
     sphinxcontrib-serializinghtml 1.1.4                      py_0    conda-forge
     sphinxcontrib-websupport  1.2.2              pyh9f0ad1d_0    conda-forge
     sqlite                    3.30.1               hcee41ef_0    conda-forge
     tblib                     1.6.0                      py_0    conda-forge
     terminado                 0.8.3            py37hc8dfbb8_1    conda-forge
     testpath                  0.4.4                      py_0    conda-forge
     thrift-cpp                0.12.0            hf3afdfd_1004    conda-forge
     tk                        8.6.10               hed695b0_0    conda-forge
     toolz                     0.10.0                     py_0    conda-forge
     tornado                   6.0.4            py37h8f50634_1    conda-forge
     traitlets                 4.3.3            py37hc8dfbb8_1    conda-forge
     typed-ast                 1.4.1            py37h516909a_0    conda-forge
     typing_extensions         3.7.4.2                    py_0    conda-forge
     uriparser                 0.9.3                he1b5a44_1    conda-forge
     urllib3                   1.25.9                     py_0    conda-forge
     wcwidth                   0.1.9              pyh9f0ad1d_0    conda-forge
     webencodings              0.5.1                      py_1    conda-forge
     wheel                     0.34.2                     py_1    conda-forge
     wrapt                     1.12.1           py37h8f50634_1    conda-forge
     xz                        5.2.5                h516909a_0    conda-forge
     yaml                      0.2.4                h516909a_0    conda-forge
     zeromq                    4.3.2                he1b5a44_2    conda-forge
     zict                      2.0.0                      py_0    conda-forge
     zipp                      3.1.0                      py_0    conda-forge
     zlib                      1.2.11            h516909a_1006    conda-forge
     zstd                      1.4.3                h3b9ef0a_0    conda-forge
     
</pre></details>

**Comment by harrism** [comment]
@afender can you try with the latest from branch-0.14 with the fix from #5193?

**Comment by afender** [comment]
Fixed :+1: Thanks.
